site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -468,27 +468,58 @@ public function showStatus($gradeable_id) {
                 ];
             }
             if ($peer) {
-                $sections['stu_grad'] = [
-                    'total_components' => $num_components * $peers_to_grade,
-                    'graded_components' => $my_grading,
-                    'num_gradeables' => $num_gradeables,
-                    'ta_graded_components' => 0,
-                    'graders' => []
-                ];
-                $sections['all'] = [
-                    'total_components' => 0,
-                    'graded_components' => 0,
-                    'graders' => []
-                ];
-                foreach ($total_users as $key => $value) {
-                    if ($key == 'NULL') {
-                        continue;
+                 // If not a team assignment => Team Peer Grading Stats Should be Visible
+                if ($gradeable->isTeamAssignment()) {
```

Comment:
This if statement is odd to me. A lot of code is repeated in both clauses (the foreach loop for example). I also don't understand why the total_components is just getPeerComponents for a team assignment: shouldn't it still be # of components times # of teams to grade?

Reply:
Yes, I tried fixing Peer stats when Teams are used but I found teams still needs to have a better way around peer assignments. [I think there we might need additional columns in some of the relations ( so that we can use joins aptly)]
It looked out of scope of this PR (and this already looks complex) so I have disabled peer view team gradeable stats while non peer view team gradeable stats are working fine. 

[Maybe we should talk more about it during meeting] 
Apart from this, I have worked on the fixes you suggested.